build-test.cmd

```diff
@@ -388,7 +389,10 @@ if errorlevel 1 (
     exit /b 1
 )
 
-if defined __SkipManaged exit /b 0
+if defined __SkipManaged (
+    rd /s /q "%CORE_ROOT_STAGE%"
```

Comment:
I wonder why deleting %CORE_ROOT_STAGE% depends on __SkipManaged?

Reply:
crossgen is called from %CORE_ROOT_STAGE% a few lines below. When skipping managed tests the script doesn't reach that point, so we need to handle the leftover staging folder.

Reply:
Do we assume that running `build-test.cmd crossgen skipmanaged` is undefined behavior?

Reply:
Changed to a `goto`